### PR TITLE
Task/887 story impement new uiux

### PR DESCRIPTION
**Pull request by tmassuia-sw**

### CODE REVIEW

File: packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.module.scss [hunk 1]
```diff
@@ -83,16 +72,15 @@
...
-  position: absolute;
-  height: 1.5em;
-  width: 1.5em;
-  z-index: 101;
-  top: 3.5rem;
-  left: 2.6rem;
+
+  height: 2.5em;
+  width: 2.5em;
+  z-index: 101;
```
Comment (dsanchez-sw) [inline packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.module.scss]: Is this z-index necessary?

File: packages/apps/web/src/components/Layout/Timeline/Timeline.module.scss [hunk 2]
```diff
@@ -142,6 +142,21 @@
 }
 
 // Contributors
+.ownerStar {
+	padding: 4px;
+	height: 1.5rem;
+	width: 1.5rem !important;
+	margin: 0px;
+	margin-left: -15px;
+	background-color: #E8B300;
```
Comment (dsanchez-sw) [inline packages/apps/web/src/components/Layout/Timeline/Timeline.module.scss]: Use $colorWarning variable.

File: packages/ui-components/src/StoryTeller/StoryTeller.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import styles from './StoryTeller.module.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faScroll } from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import RoutingContext from 'contexts/Routing/RoutingContext';
+
+import { ProfileImage } from '@timeswan/ui-components';
+
+const StoryTeller = (props) => {
```
Comment (dsanchez-sw) [inline packages/ui-components/src/StoryTeller/StoryTeller.js]: This component could have a more informative name:
Maybe **storytellerChip** or something more generic like **profileChip**, if that's what we intend.

Comment (dsanchez-sw) [reply packages/ui-components/src/StoryTeller/StoryTeller.js]: (this isn't something we should change right now)

File: packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js [hunk 4]
```diff
@@ -54,6 +55,29 @@ const StoryContributors = (props) => {
...
+				return (<> {profile.firstName} <FontAwesomeIcon icon={faFeatherAlt} /> </>);
+			}
+			else return profile.firstName;
+		}
+		else return t('UI.ME');
+	};
+
+	const resolveOwnerNameText = (profile) => {
+		if (!storyContext.isMyStory) {
+			if (profile.type !== PROFILE_TYPE.MY_LIFE) return (<> {profile.firstName} <FontAwesomeIcon icon={faFeatherAlt} /> </>);
```
Comment (frey-sw) [inline packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js]: Use cruly braces to improve reading

File: packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js [hunk 5]
```diff
@@ -70,31 +94,46 @@ const StoryContributors = (props) => {
...
+					:
+					<>
+						<div className={['col-xs-12', styles.contributorsItem].join(' ')}>
+							<ProfileImage profile={props.profile} />
+							<div className={styles.contributorsItem__data}>
+								<p className={styles.contributorName}>{storyContext.contextProfile.firstName} <FontAwesomeIcon icon={faFeatherAlt} /></p>
+								<ContributorChip contributorType={STORY_CONTRIBUTOR_TYPE.ANCESTOR} />
+							</div>
+						</div>
+					</>
```
Comment (frey-sw) [inline packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js]: The if / else must be placed inside <div className={styles.contributorsItem__data}>

File: packages/apps/web/src/components/Layout/Story/Components/StoryHeader/StoryHeader.js [hunk 6]
```diff
@@ -29,9 +28,15 @@ const StoryHeader = (props) => {
 				return t('CREATE_STORY.PAGE_TITLE');
 		}
 	};
+
+	const resolveProfileImg = () => {
+		if (storyContext.storyPageType === STORY_PAGES.NEW) return storyContext.contextProfile;
+		else return storyContext.storyData.profile;
```
Comment (frey-sw) [inline packages/apps/web/src/components/Layout/Story/Components/StoryHeader/StoryHeader.js]: Sentence can be shortened:
return storyContext.storyPageType === STORY_PAGES.NEW ? storyContext.contextProfile : storyContext.storyData.profile

File: packages/core/src/queries/Story/VIEW_STORY_QUERY.js [hunk 7]
```diff
@@ -23,7 +23,14 @@ export default gql`
                 id
                 path
             }
-            profile{ ...getStoryDataProfileData }
+            profile{ 
+							...getStoryDataProfileData
+								user { 
+									mainProfile {
+										...getStoryDataProfileData
```
Comment (frey-sw) [inline packages/core/src/queries/Story/VIEW_STORY_QUERY.js]: Is it required to fetch all the fragmentData? can we just use less info to fetch?

File: packages/ui-components/src/DropdownContainer/DropdownContainer.js [hunk 8]
```diff
@@ -43,14 +47,29 @@ const DropdownContainer = (props) => {
 
 	return (
 		<div className={styles.dropdownCont}>
-			<button type='button' className={props.buttonClassName + ' unstyledButton'} onClick={clickHandler} ref={dropdownRef} aria-haspopup='true' aria-owns={props.dropdownID} aria-label={props.buttonLabel} title={props.buttonLabel}>
+			<button
+			 type='button'
+			 className={props.buttonClassName + ' unstyledButton'}
+			 onClick={clickHandler}
+			 ref={dropdownRef}
```
Comment (frey-sw) [inline packages/ui-components/src/DropdownContainer/DropdownContainer.js]: Ref shouldnt be here

File: packages/ui-components/src/MainTitle/MainTitle.js [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'; 
 import styles from './MainTitle.module.scss';
 import PropTypes from 'prop-types';
+import { ProfileImage } from '@timeswan/ui-components';
```
Comment (frey-sw) [inline packages/ui-components/src/MainTitle/MainTitle.js]: @timeswan/components cannot be importing itself. Fix import

File: packages/ui-components/src/StoryTeller/StoryTeller.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
+import React, { useContext } from 'react';
+import styles from './StoryTeller.module.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faScroll } from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import RoutingContext from 'contexts/Routing/RoutingContext';
+
+import { ProfileImage } from '@timeswan/ui-components';
```
Comment (frey-sw) [inline packages/ui-components/src/StoryTeller/StoryTeller.js]: Same as before... ui-components shouldnt be invoking itself as imported package

File: packages/ui-components/src/StoryTeller/StoryTeller.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
+import React, { useContext } from 'react';
```
Comment (frey-sw) [inline packages/ui-components/src/StoryTeller/StoryTeller.js]: Is this component a TimeSwan exclusive or is it possible to be reutilizable? @dsanchez-sw

File: packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js [hunk 12]
```diff
@@ -54,6 +55,29 @@ const StoryContributors = (props) => {
 		props.updateContributors([...state.contributors, ...contributors]);
 	};
 
+	const resolveContributorType = (contributor) => {
+		if (contributor.id === storyContext.storyMainProfile.id) return STORY_CONTRIBUTOR_TYPE.STORYTELLER;
+		return STORY_CONTRIBUTOR_TYPE.CONNECTION;
```
[the remark this answers is not in the document]
Comment (tmassuia-sw) [reply packages/apps/web/src/components/Layout/Story/Components/StoryContributors/StoryContributors.js]: Sorry, do you have some suggestions?

File: packages/ui-components/src/DropdownContainer/DropdownContainer.js (hunk 8 above)
Comment (tmassuia-sw) [reply packages/ui-components/src/DropdownContainer/DropdownContainer.js]: @dsanchez-sw do you have some suggestion to resolve it? If we just remove the application crash.

File: packages/apps/web/src/components/Layout/Story/ViewStory/ViewStory.js [hunk 13]
```diff
@@ -1,236 +1,242 @@
...
+import { useTranslation } from 'react-i18next';
+import { faUserPlus, faPen, faSortDown, faFeatherAlt, faScroll, faBookmark } from '@fortawesome/free-solid-svg-icons';
+import Fade from 'react-reveal/Fade';
+import styles from './ViewStory.module.scss';
+
+import StoryContext from 'contexts/Story/StoryContext';
+import ModalsContext from 'contexts/Modals/ModalsContext';
+import RoutingContext from 'contexts/Routing/RoutingContext';
+import Comments from '../Components/Comments/Comments';
+import ReturnFooter from '../../../../UI/ReturnFooter/ReturnFooter';
```
Comment (frey-sw) [inline packages/apps/web/src/components/Layout/Story/ViewStory/ViewStory.js]: Change To import ReturnFooter from 'UI/ReturnFooter/ReturnFooter';

File: packages/apps/web/src/components/Layout/Timeline/Timeline.js [hunk 14]
```diff
@@ -1,541 +1,548 @@
...
+	return (
+		loading ? <Overlay type={OVERLAY_TYPES.LOADING} />
+			: <section className='container'>
+				<div>
+					<div className={styles.timelineTitleCont}>
+						<MainTitle
+							heading={TITLE_HEADING.H1}
+							title={state.isMyTimeline && !ancestorProfile ?
+								t('TIMELINE.TITLE') :
+								t('TIMELINE.VISITOR_TITLE', { firstName: (data ? data.profile.firstName : null) })}
```
Comment (frey-sw) [inline packages/apps/web/src/components/Layout/Timeline/Timeline.js]: Move this to a function

**Pull request merged**